### ISSUE TITLE
wandb: latest step's metrics only flush when the next step logs (explicit-step commit lag)

### ISSUE THREAD

**Issue by shuheng-liu** [TensorAuto/OpenTau#353]
## Symptom

A step's metrics (and especially eval results + videos) don't appear on the wandb dashboard until the **next** logging step fires. With `eval_freq=1000` and ~20 min between training logs, you finish a multi-minute eval and then wait ~20 min — until the next training step logs — before the eval point and its videos show up in the workspace. The media files appear in the **Files** tab promptly (eager upload), but the **charts/panels** lag by one logging interval.

Two consequences:
1. **Display lag**: the newest logged step always trails by one interval. Cosmetic during training, but annoying for eval — you can't see the eval result right after it runs.
2. **Ungraceful-exit loss**: if the run is killed (e.g. `scancel` without a graceful `wandb.finish()`) the final *pending* step can be missing from the server even though it's in the local `.wandb` transaction log. (Recoverable with `wandb sync <run_dir>`, but easy to miss.)

## Root cause

`scripts/train.py` logs with an **explicit `step=`** and makes ~20 separate `accelerator.log({single_key: val}, step=step)` calls per step — training block (`train.py:684-693`), validation block (`train.py:790-812`), eval block (`train.py:869-890`). None pass a `commit` kwarg.

When you call `wandb.log(data, step=N)` with an explicit step, wandb does **not** seal step N's row immediately — it keeps N as the "current pending" row so that further `log(..., step=N)` calls can accumulate more keys into the same row (which OpenTau relies on — that's how 20 separate calls land in one row). wandb only finalizes + flushes row N when it sees a `log()` with `step > N`, or on `wandb.finish()`, or a periodic flush. So row N is committed when N+1's first log call arrives.

## Mechanism for a fix (verified)

`accelerate` 1.12's `WandBTracker.log(values, step, **kwargs)` forwards `**kwargs` straight to `wandb.run.log(values, step=step, **kwargs)`, and `Accelerator.log(values, step, log_kwargs={})` passes `log_kwargs[tracker.name]` to each tracker. So a `commit=True` can be threaded through, tracker-safe (other trackers never see it):

```python
accelerator.log(values, step=step, log_kwargs={"wandb": {"commit": True}})
```

## Proposed fix

Add a single **explicit commit point per logging step**, placed *after* all conditional logging blocks (training + validation + eval) for that step, so the row is sealed and flushed as soon as everything for step N is logged — not when N+1 arrives.

Minimal shape: a small helper invoked unconditionally at the bottom of the per-step logging section:

```python
def _commit_wandb_step(accelerator, step):
    # Seal + flush the pending wandb row for `step` immediately, instead of
    # waiting for the next step's first log() to auto-commit it. Tracker-safe:
    # non-wandb trackers ignore log_kwargs they aren't keyed for.
    accelerator.log({}, step=step, log_kwargs={"wandb": {"commit": True}})
```

(Verify `log({}, commit=True)` reliably commits the pending row; if an empty dict is a no-op in some wandb version, fold `commit=True` into the genuinely-last metric log of the step instead.)

### Important caveat — commit must be last

After step N is committed, **no further `log(..., step=N)` may be issued** — wandb requires monotonically increasing steps and will warn-and-drop a late write to a sealed step. So the commit must come after every training/validation/eval log for that step. The blocks fire at different cadences (`log_freq` / `val_freq` / `eval_freq`), and when several coincide the last block varies, so a single unconditional flush at the very bottom of the logging section is safer than tacking `commit=True` onto one specific call.

## Alternative considered

Consolidate the ~20 per-section `accelerator.log({single_key: val}, ...)` calls into one dict per section (`accelerator.log({...all training metrics...}, step=step)`). Worth doing for clarity/overhead regardless, but it does **not** by itself fix the lag — a single explicit-step `log()` still holds the row open until the next step. The commit point is the actual fix; the consolidation is a nice-to-have that pairs well with it.

## Acceptance criteria

- After an eval at step N completes, its `Success Rate` / `Success/<group>` scalars and `Eval Videos/*` panels appear on the dashboard **without** waiting for step N+1.
- The final step before a clean shutdown is present on the server.
- x-axis (`_step`) still aligns with the training step for all metric families (training/validation/eval), and no "step must be monotonically increasing" warnings appear in the logs.
- Non-wandb trackers (if configured) are unaffected.

## Notes

Low priority / quality-of-life — no correctness impact on training itself; purely about observability latency and resilience of the last logged step. Surfaced while watching a long pretrain run: each eval's results lagged ~20 min behind the eval actually finishing.
